ftl/node/builder_test.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import unittest
+import tarfile
+import cStringIO
+import mock
+import httplib2
+
+from ftl.common import context
+from ftl.common import cache
+
+from ftl.node import builder
+
+
+class NodeTest(unittest.TestCase):
+
+    def test_create_package_base_uncached(self):
+        # Things to test for node:
+        # for uncached
+        b = builder.Node(None)
+        base_img = mock.MagicMock()
+        files = {
+            'packages.json': """{
+  "name": "hello-world-express",
+  "description": "Hello World test app",
+  "version": "0.0.1",
+  "private": true,
+  "dependencies": {
+    "express": "3.x"
+  }
+}
+""",
+            'app.js': """var express = require('express');
+var app = express();
+
+// Routes
+app.get('/', function(req, res) {
+  res.send('Hello World!');
+});
+
+// Listen
+var port = process.env.PORT || 3000;
+app.listen(port);
+console.log('Listening on localhost:'+ port);
+""",
+        }
+        # check that these files are in the tar
+        # check that express was installed into node_modules
+        # check that layer was cached
+
+        # create a mock docker registry for the cache (disk/memory cache?)
```

Comment:
I think that I can actually do something very similar to what is already done for the JustApp tests using the memory context.  The main difference is verifying that "express" (and potentially other packages) are installed via npm for node.